File: python/leap/leap.py
def is_leap_year(year):
    div_4 = year % 4 == 0
    div_100 = year % 100 == 0
    div_400 = year % 400 == 0
    if not div_4:
        return False
    if div_400:
        return True
    if div_100:
        return False
    return True

File: go/leap/leap.go
package leap

const (
	testVersion   = 3
	rule_4_year   = 4
	rule_100_year = 100
	rule_400_year = 400
)

func IsLeapYear(year int) bool {
	if year%rule_4_year == 0 {
		if year%rule_100_year == 0 {
			if year%rule_400_year == 0 {
				return true
			} else {
				return false
			}
		} else {
			return true
		}
	} else {
		return false
	}
}
